src/wdio/services/SeleniumDockerService.js

```diff
@@ -65,39 +31,130 @@ export default class SeleniumDockerService {
     this.port = config.port;
     this.path = config.path;
     this.browserName = capabilities[0].browserName;
-    this.cidfile = this.config.cidfile || this.cidfile;
 
+    if (!this.config.enabled) {
+      return;
+    }
+
+    this.container = await this.getRunningContainer();
+
+    // There is a running container that doesn't match configuration, stop it
+    // before proceeding
+    if (this.container && this.getImage() !== this.container.image) {
+      await this.stop();
+
+    // There is a container, verify selenium is running as expected
+    } else if (this.container) {
+      try {
+        await this.ensureSelenium();
+
+      // Selenium isn't responding stop the container;
+      } catch (e) {
+        await this.stop();
+      }
+    }
+
+    if (!this.container) {
+      await this.pullImage();
+      await this.runImage();
+      await this.ensureSelenium();
+    }
+  }
+
+  /**
+   * Clean up docker container after all workers got shut down and the process is about to exit.
+   */
+  async onComplete() {
+    if (this.config.cleanup && this.config.enabled) {
```

Comment:
Do we need to check `this.config.enbabled`? above if not enabled its not kicking it off.

Reply:
the onComplete will still be called and we don't want to run the docker command if its not enabled.